### PR TITLE
Improve button response

### PR DESCRIPTION
**Pull request by darinfranklin**
Reformat the main loop with one statement per line so that future changes will be easier to diff.
Produce keypress events in the ISR to separate the button press detection from the event handling in the main loop.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 SDCC ?= sdcc
 STCCODESIZE ?= 4089
-SDCCOPTS ?= --iram-size 256 --code-size $(STCCODESIZE) --xram-size 0 --data-loc 0x30 --disable-warning 126 --disable-warning 59
+SDCCOPTS ?= --iram-size 256 --code-size $(STCCODESIZE) --xram-size 0 --data-loc 0x30 --disable-warning 126 --disable-warning 59 \
+    -DDEBUG -DWITH_ALT_LED9 -DWITHOUT_LEDTABLE_RELOC
```
Comment (zerog2k) [inline Makefile]: Are/should these be exposed as make build options so user can do something like `make WITH_ALT_LED9=1` etc?
Another option for the future might be to consider individual make targets at least for the board/model variations, and leverage the make build options for various features.
Anyhow, going to merge this now, as it seems to work fine as is on my stc15f204ea version (with two buttons).

**Comment by zerog2k** [comment]
great job, thanks for the PR. Buttons are indeed much more responsive.

File: Makefile (hunk 1 above)
Comment (darinfranklin) [inline Makefile]: Yes, I would like to add more optional features, so I will figure out how to specify these without editing the Makefile.  Thanks for accepting the PR.

Comment (zerog2k) [inline Makefile]: Ok here's one example of one way to parse make options. 
https://github.com/BastilleResearch/nrf-research-firmware/pull/8/files

**Pull request merged**